docs/serving/samples/hello-world/helloworld-go/index.md

```diff
@@ -5,4 +5,212 @@ weight: 1
+   RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly -v -o server
+
+   # Use the official Alpine image for a lean production container.
+   # https://hub.docker.com/_/alpine
+   # https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+   FROM alpine:3
+   RUN apk add --no-cache ca-certificates
+
+   # Copy the binary to the production image from the builder stage.
+   COPY --from=builder /app/server /server
+
+   # Run the web service on container startup.
+   CMD ["/server"]
+   ```
+
+1. Use the go tool to create a
+   [`go.mod`](https://github.com/golang/go/wiki/Modules#gomod) manifest.
+
+   ```shell
+   go mod init github.com/knative/docs/docs/serving/samples/hello-world/helloworld-go
+   ```
+
+## Building and deploying the sample
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+1. Use Docker to build the sample code into a container. To build and push with
+   Docker Hub, run these commands replacing `{username}` with your Docker Hub
+   username:
+
+   ```shell
+   # Build the container on your local machine
+   docker build -t {username}/helloworld-go .
+
+   # Push the container to docker registry
+   docker push {username}/helloworld-go
+   ```
+
+1. After the build has completed and the container is pushed to docker hub, you
+   can deploy the app into your cluster.  Pick one of the following methods:
+
+
+   {{< tabs name="helloworld_go" default="kn" >}}
+   {{% tab name="yaml" %}}
+
+   Create a new file, `service.yaml` and copy the following service definition
+   into the file. Make sure to replace `{username}` with your Docker Hub
+   username.
+
+   ```yaml
+   apiVersion: serving.knative.dev/v1
+   kind: Service
+   metadata:
+     name: helloworld-go
+     namespace: default
+   spec:
+     template:
+       spec:
+         containers:
+           - image: docker.io/{username}/helloworld-go
+             env:
+               - name: TARGET
+                 value: "Go Sample v1"
+   ```
+
+   Ensure that the container image value
+   in `service.yaml` matches the container you built in the previous step. Apply
+   the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+   Now that your service is created, Knative will perform the following steps:
+
+   - Create a new immutable revision for this version of the app.
+   - Network programming to create a route, ingress, service, and load balance
+     for your app.
+   - Automatically scale your pods up and down (including to zero active pods).
+
+   Run the following command to find the domain URL for your service:
+
+   ```shell
+   kubectl get ksvc helloworld-go  --output=custom-columns=NAME:.metadata.name,URL:.status.url
+   ```
+
+   Example:
+
+   ```shell
+    NAME                URL
+    helloworld-go       http://helloworld-go.default.1.2.3.4.xip.io
+   ```
```

Comment:
No. 4; also I feel like this should be a verification step instead.
I'm not sure that tabs are the best way to display this info at all tbh...

Reply:
What do you suggest?  We'd been talking about tabs for a while, which is why I'd bit the bullet here.  Definitely open to suggestions.